lib/elixir/lib/enum.ex

```diff
@@ -803,11 +803,13 @@ defmodule Enum do
   @spec filter_map(t, (element -> as_boolean(term)),
     (element -> element)) :: list
 
-  def filter_map(enumerable, filter, mapper) when is_list(enumerable) do
+  def filter_map(enumerable, filter, mapper)
+  when is_list(enumerable) and is_function(filter, 1) and is_function(mapper, 1) do
```

Comment:
the only drawback is that if you end up with odd number of spaces, you cannot use the TAB key to indent it to that level.